follow_fork/follow_fork.c

```diff
@@ -0,0 +1,123 @@
+#define _POSIX_SOURCE
+#define _GNU_SOURCE
+
+#include "undolr.h"
+#include "undolr_deprecated.h"
+
+#include <errno.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdbool.h>
+#include <time.h>
+#include <stdlib.h>
+#include <dlfcn.h>
+#include <linux/limits.h>
+
+extern char *__progname;
+
+void attach_lr (int, char**, char**) __attribute__ ((constructor));
+
+static bool
+is_valid_process(pid_t pid)
+{
+    if (strncmp(__progname, "undo", 4) == 0)
+    {
+        if (unsetenv("LD_PRELOAD") == -1)
+        {
+            perror("unsetenv");
+            exit(1);
+        }
+        return false;
+    }
+    return true; 
+}
+
+static void*
+get_sym_addr(const char *sym)
+{
+    void *fptr = NULL;
+    
+    fptr = dlsym(RTLD_NEXT, sym);
+    if (fptr == NULL)
+    {
+        fprintf(stderr, "Error: %s, calling dlsym with %s\n",
+                        dlerror(), sym);
+        abort();
+    }
+    return fptr;
+}
+
+static bool
+start_recording(pid_t pid)
+{
+    undolr_error_t err;
+    int e = undolr_start(&err);
+    if (e)
+    {
+        fprintf(stderr, "undolr_recording_start failed: err=%i, errno=%i\n", err, errno);
```

Comment:
You should do this to give a better error message:
```
    if (e)
    {
        perror("undolr_start");
        fprintf(stderr, "%s\n", undolr_error_string(error));
        return false;
    }
```

Reply:
fair enough, fixed